### PR TITLE
chore: Test that `go mod tidy` is a no-op in PR CI run

### PR DESCRIPTION
**Pull request by 0x656b694d**
CI

### CODE REVIEW

**Review by janisz — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -104,6 +104,9 @@ jobs:
           key: ${{ runner.os }}-go-${{ hashFiles('go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go-
+      - name: Test go modules
+        run: |
+          go mod tidy && git diff --exit-code
```
Comment (janisz) [inline .github/workflows/ci.yaml:109]: IMO this should be a precommit check

Comment (janisz) [reply .github/workflows/ci.yaml:109]: https://pre-commit.com/hooks.html#:~:text=go%2Dmod%2Dtidy%20%2D%20Runs%20%60go%20mod%20tidy%20%2Dv%60%2C%20requires%20golang

Comment (porridge) [reply .github/workflows/ci.yaml:109]: I think its first run is a little to slow for pre-commit :thinking:

```
[mowsiany@mowsiany stackrox]$ time go mod tidy
go: downloading github.com/frankban/quicktest v1.14.6
go: downloading github.com/cncf/xds/go v0.0.0-20231109132714-523115ebc101
go: downloading github.com/go-git/go-git-fixtures/v4 v4.3.2-0.20231010084843-55a94097c399
go: downloading github.com/elazarl/goproxy v0.0.0-20230808193330-2592e75ae04a

real	0m24.550s
user	0m11.498s
sys	0m15.043s
[mowsiany@mowsiany stackrox]$ time go mod tidy

real	0m1.960s
user	0m3.922s
sys	0m3.007s
[mowsiany@mowsiany stackrox]$ 
```

**Review by porridge — APPROVED** [review]

**Comment by 0x656b694d** [comment]
Sorry, the change merged within #1620.

**Pull request closed without merging**